### PR TITLE
203 pass mf2 wg syntax tests

### PR DESCRIPTION
**Pull request by mvilks**

### CODE REVIEW

**Comment by mvilks** [comment]
Unknown-function errors handled.
Number literal formatting gives headache.

File: template/template.go [hunk 1]
```diff
@@ -355,6 +355,10 @@ func (e *executer) resolveExpression(expr ast.Expression) (*ResolvedValue, error
 
 		switch v := expr.Operand.(type) {
 		default:
+			if f, ok := expr.Annotation.(ast.Function); ok {
+				return NewResolvedValue("{:" + f.Identifier.String() + "}")
```
Comment (janishorsts) [inline template/template.go:359]: What about a helper function _newFallbackValue_? We return fall-back value in many places.

File: mf2wg_test.go [hunk 2]
```diff
@@ -255,14 +236,16 @@ func assertErr(t *testing.T, want Errors, err error) {
 			wantErr(mf2.ErrMissingFallbackVariant)
 		case "missing-selector-annotation":
 			wantErr(mf2.ErrMissingSelectorAnnotation)
+		case "syntax-error":
```
Comment (janishorsts) [inline mf2wg_test.go:239]: please ensure case are ORDERED by name

Comment (mvilks) [reply mf2wg_test.go:239]: yes, that's why I reordered them

File: template/template.go (hunk 1 above)
Comment (mvilks) [reply template/template.go]: Implemented `newFallbackValue` that does all the type checking and correct wrapping.

File: template/template.go [hunk 3]
```diff
@@ -109,6 +109,39 @@ func NewResolvedValue(value any, options ...ResolvedValueOpt) *ResolvedValue {
 	return r
 }
 
+func newFallbackValue(value any) *ResolvedValue {
```
Comment (janishorsts) [inline template/template.go:112]: I'd expect it to implement https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md#fallback-resolution and nothing else.

**Review by janishorsts — APPROVED** [review]

**Pull request merged**